### ISSUE TITLE
Renaming a new tag before saving creates unusable tag

### ISSUE THREAD

**Issue by evanostroski** [HabitRPG/habitica#7806]
General Info
- UUID: aecf6364-1cfb-4b5c-98c0-241b88b2e0d8
- Browser: Chrome Version 51.0.2704.103 (64-bit)
- OS: Darwin 407s-MacBook-Pro.local 15.5.0 Darwin Kernel Version 15.5.0: Tue Apr 19 18:36:36 PDT 2016; root:xnu-3248.50.21~8/RELEASE_X86_64 x86_64
### Description

Renaming a tag before saving creates an unusable tag
#### Steps to Reproduce
1. Click Tag **Edit**
2. Type `test1` into new Tag
3. Click **Add**
4. Change Tag `test1` to `test2`
5. Click **Save** (`test2` appears in Tag list)
6. Create a new Task
7. Edit the new Task
8. Check the box next to `test2`
9. Error

<img width="271" alt="screen shot 2016-07-18 at 12 34 02 pm" src="https://cloud.githubusercontent.com/assets/2172501/16928107/c6af8e54-4ce5-11e6-8629-4c56fc345820.png">

Here's a network capture of the failed request and of the tag creation:
[tag rename bug.har.zip](https://github.com/HabitRPG/habitrpg/files/369980/tag.rename.bug.har.zip)
#### Console Errors

```
app-bbe0d239.js:8 
POST https://habitica.com/api/v3/tasks/c696eb10-fb33-4fb9-b979-fb4ec0708660/tags/4368aab4-63f0-4920-b712-e450d0a944c6?userV=79 400 (Bad Request)

(anonymous function) @ app-bbe0d239.js:8
s @ app-bbe0d239.js:8
f @ app-bbe0d239.js:8
j @ app-bbe0d239.js:9
(anonymous function) @ app-bbe0d239.js:9
$eval @ app-bbe0d239.js:9
$digest @ app-bbe0d239.js:9
$apply @ app-bbe0d239.js:9
(anonymous function) @ app-bbe0d239.js:10
dispatch @ app-bbe0d239.js:4
r.handle @ app-bbe0d239.js:4
```

**Comment by Ozzystrasza** [comment]
I got that error too when I tried to check a new tag I created into a task. But it worked once I refreshed the page.

**Comment by evanostroski** [comment]
Think the issue is in : /website/client/js/controllers/filtersCtrl.js  
I'll take a look when I get the chance.

**Comment by JSchneidler** [comment]
I believe this is related to #7820 and should be fixed by a PR I submitted. Either way, I think the root cause is the same and this could be closed and continued in #7820?

**Comment by Alys** [comment]
I've added this to the top post in #7820 :

The fix for this should also fix https://github.com/HabitRPG/habitrpg/issues/7806 or if that's not possible, https://github.com/HabitRPG/habitrpg/issues/7806 should be reopened.
